tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -489,6 +490,7 @@ StorageManager::array_reopen(
       return {st, std::nullopt, std::nullopt, std::nullopt};
 
     open_array = it->second;
+    lock.unlock();
```

Comment:
Could you just move the closing bracket up here, when the lock goes out of scope, it will get unlocked.